components/builder-depot/src/server.rs

```diff
@@ -1241,6 +1241,73 @@ fn list_package_versions(req: &mut Request) -> IronResult<Response> {
     }
 }
 
+fn list_origin_packages_for_version(req: &mut Request) -> IronResult<Response> {
+
+    let (start, stop) = match extract_pagination(req) {
+        Ok(range) => range,
+        Err(response) => return Ok(response),
```

Comment:
That's a great question! I raised the question here solely on the basis of just reading this code; it read a bit oddly to me. If this is a pattern used elsewhere in this module, it may simply be a matter of renaming things to better reflect what's going on (i.e., make it read better). I'd need to dig a bit further to know for sure (I haven't touched this code yet!) On the other hand, based on @raskchanky 's feedback below about how we are using an HTTP Partial Content response for paginated responses, there may be more extensive refactorings that are worthwhile. This PR may not be the place for such a fix.